test/integration-ebpf/src/bpf_probe_read.rs

```diff
@@ -72,8 +72,13 @@ pub fn test_bpf_probe_read_kernel_str_bytes(ctx: ProbeContext) {
     );
 }
 
-#[cfg(not(test))]
+#[cfg(target_arch = "bpf")]
 #[panic_handler]
 fn panic(_info: &core::panic::PanicInfo) -> ! {
     loop {}
 }
+
+#[cfg(not(target_arch = "bpf"))]
+fn main() {
+    panic!("This should only ever be called from its eBPF entrypoint")
```

Comment:
let's use this message to explain why this function exists - and please write a meaningful explanation, i.e. not "because no_main is cfg-ed out" but something that explains why no_main must be cfg'ed out.